### PR TITLE
Refactored Examples

### PR DESCRIPTION
**Pull request by WilliamDenniss**
– Moved all examples into Examples/ folder.
– Removed all Google-specific information from source code and generic READMEs.
– Added a Google-specific read, and instructions for other OpenID Certified providers to do the same.

Per the [announcement](http://lists.openid.net/pipermail/openid-specs-ab/Week-of-Mon-20170306/006456.html) on-list.

---
This change is [Reviewable](https://reviewable.io/reviews/openid/appauth-ios/80)

### CODE REVIEW

File: Examples/Example-macOS/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Example Project
```
Comment (iainmcgin) [inline Examples/Example-macOS/README.md]: Same comments for this file apply as those for the iOS readme

File: Examples/README-Google.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Using AppAuth for iOS and macOS with Google
+
+To configure the sample with a Google OAuth client, first visit
```
Comment (iainmcgin) [inline Examples/README-Google.md]: This page is also excellent for quickly configuring an iOS client:

https://developers.google.com/mobile/add?platform=ios

File: Examples/Example-iOS_ObjC/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Setup & Open the Project
+
+Open the `Example.xcworkspace` file. This contains the example app,
+and embeds the AppAuth dependency. You can also use Pods to manage the
+dependency, an example Podfile is contained in the Example-macOS folder.
+
+## Configuration
+
+The example doesn't work out of the box, you need to configure it your own
```
Comment (WilliamDenniss) [inline Examples/Example-iOS_ObjC/README.md]: Done

File: Examples/Example-iOS_ObjC/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+client ID.
+
+### Information You'll Need
+
+* Issuer
+* Client ID
+* Redirect URI
+
+The instructions on how to get this vary by IdP, but we have a
```
Comment (WilliamDenniss) [inline Examples/Example-iOS_ObjC/README.md]: Done

Comment (WilliamDenniss) [inline Examples/Example-iOS_ObjC/README.md]: Done

File: Examples/Example-iOS_ObjC/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+client ID.
+
+### Information You'll Need
+
+* Issuer
+* Client ID
+* Redirect URI
+
+The instructions on how to get this vary by IdP, but we have a 
+[docs](../README.md#openid-certified-providers) for some OpenID Certified
```
Comment (WilliamDenniss) [inline Examples/Example-iOS_ObjC/README.md]: Done

File: Examples/Example-macOS/README.md (hunk 1 above)
Comment (WilliamDenniss) [inline Examples/Example-macOS/README.md]: Done

File: Examples/README-Google.md (hunk 2 above)
Comment (WilliamDenniss) [inline Examples/README-Google.md]: Per our conversation, will keep as-is for now.

**Comment by iainmcgin** [comment]
LGTM

**Pull request merged**